Content.Server/Instruments/InstrumentSystem.cs

```diff
@@ -66,6 +65,28 @@ public override void Shutdown()
         ShutdownCVars();
     }
 
+    private void OnBoundUIClosed(EntityUid uid, InstrumentComponent component, BoundUIClosedEvent args)
+    {
+        if (HasComp<ActiveInstrumentComponent>(uid)
+            && _userInterfaceSystem.TryGetUi(uid, args.UiKey, out var bui)
+            && bui.SubscribedSessions.Count == 0)
+        {
+            RemComp<ActiveInstrumentComponent>(uid);
+        }
+
+        Clean(uid, component);
+    }
+
+    private void OnBoundUIOpened(EntityUid uid, InstrumentComponent component, BoundUIOpenedEvent args)
+    {
+        if (!HasComp<ActiveInstrumentComponent>(uid))
+        {
+            EnsureComp<ActiveInstrumentComponent>(uid);
+        }
```

Comment:
Isn't EnsureComp just "if it doesn't have then add"?

Reply:
woops